core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java

```diff
@@ -120,9 +123,11 @@ public Collection<UfsStatus> addChildren(AlluxioURI path, Collection<UfsStatus>
       AlluxioURI childPath = path.joinUnsafe(child.getName());
       addStatus(childPath, child);
```

Comment:
If one of the child path and ufs status does not match, `addStatus` will throw an exception, breaking the for loop. The UFS_STATUS_CACHE_SIZE_TOTAL counter has been partially incremented, thus leaving the two counter in a inconsistent state.

Reply:
This is invalid because `childPath` is derived from `child` which is its `UfsStatus`. resolving this.